### ISSUE TITLE
Getting the names of generic types on a function.

### ISSUE THREAD

**Issue by CraigSiemens** [krzysztofzablocki/Sourcery#986]
Say I have a function like this.

```swift
func doSomething<G1, G2>(with thing: G1, and otherThing: G2) { ... }
```

I'd like to get an array of the generic types (just names or an object of some kind) that the function defines (`[G1, G2]`). `Method` has an `isGeneric` property, but there doest seem to be a way to get the generic types or know which properties are using the generic types.

My current work around has been something kind gross like this that splits the method name to get the generic type names.

```stencil
{% for genericTypeName in method.shortName | replace:", ","-" | replace:"<","-" | replace:">","" | split:"-"  %}
{% if not forloop.first %}
{# do something with genericTypeName #}
{% endif %}
{% endfor %}
```

**Comment by krzysztofzablocki** [comment]
couldn't you use `method.parameters` and just grab their `typeName` ?

**Comment by CraigSiemens** [comment]
That would work when all parameters are generic, but if there's a mix it isn't possible to tell which ones are generic and which ones aren't.

```swift
func doSomething<G1>(with thing: G1, and otherThing: Int)
```

My end goal is to make a struct like the following from any function

```
struct DoSomethingParameters<G1> {
    var thing: G1
    var otherThing: Int
}
```

**Comment by krzysztofzablocki** [comment]
makes sense, I don't think anything available for that right now, so we'd need to add it
